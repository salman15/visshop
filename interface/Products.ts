import { mediaCollectionProps, mediaSingleProps } from ".";

export interface productProps {
  id: number;
  attributes: {
    price: number;
    qty: number;
    title: string;
    description: string;
    details: string;
    thumbnail?: mediaSingleProps;
    Image_collection?: mediaCollectionProps;
  };
}
