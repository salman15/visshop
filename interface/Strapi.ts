export interface imageProps {
  name: string;
  hash: string;
  ext: string;
  mime: string;
  path: string;
  width: number;
  height: number;
  size: number;
  url: string;
}
export interface mediaProps {
  id: number;
  attributes: {
    name: string;
    alternativeText: string;
    caption: string;
    width: number;
    height: number;
    formats: {
      thumbnail: imageProps;
      medium: imageProps;
      small: imageProps;
    };
    hash: string;
    ext: string;
    mime: string;
    size: number;
    url: string;
    previewUrl: string;
    provider: string;
    provider_metadata: string;
    createdAt: string;
    updatedAt: string;
  };
}

export interface mediaSingleProps {
  data: mediaProps;
}

export interface mediaCollectionProps {
  data: mediaProps[];
}

export interface metaProps {
  pagination?: {
    page: number;
    pageSize: number;
    pageCount: number;
    total: number;
  };
}
