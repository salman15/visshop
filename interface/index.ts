export * from "./Products";
export * from "./Strapi";
