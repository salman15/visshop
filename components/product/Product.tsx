import Image from "next/image";
import React, { FC, useState } from "react";
import { productProps } from "../../interface";
import { toCurrency } from "../../utils";
interface Props {
  item: productProps;
}

const Product: FC<Props> = ({ item }) => {
  const [checked, setChecked] = useState(false)
  return (
    <div className="flex font-sans">
      <div className="min-w-12 w-full max-w-56 relative">
        {item.attributes.thumbnail && <Image
          src={`${process.env.BE_URL}${item.attributes.thumbnail.data.attributes.url}`}
          loading="lazy"
          layout="fill"
          width={200}
          height={200}
          alt=""
          className="absolute inset-0 w-full h-full object-cover rounded-lg"
        />}
      </div>
      <form className="flex-auto p-6">
        <div className="flex flex-wrap">
          <h1 className="flex-auto font-medium text-slate-900">
            {item.attributes.title}
          </h1>
          <div className="w-full flex-none mt-2 order-1 text-3xl font-bold text-violet-600">
            {toCurrency(item.attributes.price)}
          </div>
          <div className="text-sm font-medium text-slate-400">
            {item.attributes.qty}
          </div>
        </div>
        <div className="flex items-baseline mt-4 mb-6 pb-6 border-b border-slate-200">
          <div className="space-x-2 flex text-sm font-bold">
            <label>
              <input
                className="sr-only peer"
                name="size"
                type="radio"
                value="xs"
                checked={checked}
                onChange={(e) => setChecked(e.target.checked)}
              />
              <div className="w-9 h-9 rounded-full flex items-center justify-center text-violet-400 peer-checked:bg-violet-600 peer-checked:text-white">
                XS
              </div>
            </label>
            <label>
              <input
                className="sr-only peer"
                name="size"
                type="radio"
                value="s"
              />
              <div className="w-9 h-9 rounded-full flex items-center justify-center text-violet-400 peer-checked:bg-violet-600 peer-checked:text-white">
                S
              </div>
            </label>
            <label>
              <input
                className="sr-only peer"
                name="size"
                type="radio"
                value="m"
              />
              <div className="w-9 h-9 rounded-full flex items-center justify-center text-violet-400 peer-checked:bg-violet-600 peer-checked:text-white">
                M
              </div>
            </label>
            <label>
              <input
                className="sr-only peer"
                name="size"
                type="radio"
                value="l"
              />
              <div className="w-9 h-9 rounded-full flex items-center justify-center text-violet-400 peer-checked:bg-violet-600 peer-checked:text-white">
                L
              </div>
            </label>
            <label>
              <input
                className="sr-only peer"
                name="size"
                type="radio"
                value="xl"
              />
              <div className="w-9 h-9 rounded-full flex items-center justify-center text-violet-400 peer-checked:bg-violet-600 peer-checked:text-white">
                XL
              </div>
            </label>
          </div>
        </div>
        <div className="flex space-x-4 mb-5 text-sm font-medium">
          <div className="flex-auto flex space-x-4">
            <button
              className="h-10 px-6 font-semibold rounded-full bg-violet-600 text-white"
              type="submit"
            >
              Buy now
            </button>
            <button
              className="h-10 px-6 font-semibold rounded-full border border-slate-200 text-slate-900"
              role="add-to-cart"
              type="button"
            >
              Add to bag
            </button>
          </div>
          <button
            className="flex-none flex items-center justify-center w-9 h-9 rounded-full text-violet-600 bg-violet-50"
            type="button"
            aria-label="Like"
          >
            <svg width="20" height="20" fill="currentColor" aria-hidden="true">
              <path
                fillRule="evenodd"
                clipRule="evenodd"
                d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
              />
            </svg>
          </button>
        </div>
        <p className="text-sm text-slate-500">
          Free shipping on all continental US orders.
        </p>
      </form>
    </div>
  );
};

export default Product;
