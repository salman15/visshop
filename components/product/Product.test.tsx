// __tests__/fetch.test.js
import React from 'react'
import { render, fireEvent, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom'
import Product from './Product'
import { productProps } from '../../interface'

const item: productProps = {
    id: 1,
    attributes: {
        price: 10.50,
        qty: 10,
        title: 'Shrimp',
        description: 'Fresh shrimp',
        details: 'New product',
    }
}

test('Renders', async () => {
    const { container, getByRole } = render(<Product item={item} />)
    expect(getByRole('add-to-cart')).toBeTruthy()
    expect(container).toMatchSnapshot()

})