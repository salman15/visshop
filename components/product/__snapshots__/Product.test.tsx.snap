// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
<div>
  <div
    class="flex font-sans"
  >
    <div
      class="min-w-12 w-full max-w-56 relative"
    />
    <form
      class="flex-auto p-6"
    >
      <div
        class="flex flex-wrap"
      >
        <h1
          class="flex-auto font-medium text-slate-900"
        >
          Shrimp
        </h1>
        <div
          class="w-full flex-none mt-2 order-1 text-3xl font-bold text-violet-600"
        >
          € 10,5
        </div>
        <div
          class="text-sm font-medium text-slate-400"
        >
          10
        </div>
      </div>
      <div
        class="flex items-baseline mt-4 mb-6 pb-6 border-b border-slate-200"
      >
        <div
          class="space-x-2 flex text-sm font-bold"
        >
          <label>
            <input
              class="sr-only peer"
              name="size"
              type="radio"
              value="xs"
            />
            <div
              class="w-9 h-9 rounded-full flex items-center justify-center text-violet-400 peer-checked:bg-violet-600 peer-checked:text-white"
            >
              XS
            </div>
          </label>
          <label>
            <input
              class="sr-only peer"
              name="size"
              type="radio"
              value="s"
            />
            <div
              class="w-9 h-9 rounded-full flex items-center justify-center text-violet-400 peer-checked:bg-violet-600 peer-checked:text-white"
            >
              S
            </div>
          </label>
          <label>
            <input
              class="sr-only peer"
              name="size"
              type="radio"
              value="m"
            />
            <div
              class="w-9 h-9 rounded-full flex items-center justify-center text-violet-400 peer-checked:bg-violet-600 peer-checked:text-white"
            >
              M
            </div>
          </label>
          <label>
            <input
              class="sr-only peer"
              name="size"
              type="radio"
              value="l"
            />
            <div
              class="w-9 h-9 rounded-full flex items-center justify-center text-violet-400 peer-checked:bg-violet-600 peer-checked:text-white"
            >
              L
            </div>
          </label>
          <label>
            <input
              class="sr-only peer"
              name="size"
              type="radio"
              value="xl"
            />
            <div
              class="w-9 h-9 rounded-full flex items-center justify-center text-violet-400 peer-checked:bg-violet-600 peer-checked:text-white"
            >
              XL
            </div>
          </label>
        </div>
      </div>
      <div
        class="flex space-x-4 mb-5 text-sm font-medium"
      >
        <div
          class="flex-auto flex space-x-4"
        >
          <button
            class="h-10 px-6 font-semibold rounded-full bg-violet-600 text-white"
            type="submit"
          >
            Buy now
          </button>
          <button
            class="h-10 px-6 font-semibold rounded-full border border-slate-200 text-slate-900"
            role="add-to-cart"
            type="button"
          >
            Add to bag
          </button>
        </div>
        <button
          aria-label="Like"
          class="flex-none flex items-center justify-center w-9 h-9 rounded-full text-violet-600 bg-violet-50"
          type="button"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="20"
            width="20"
          >
            <path
              clip-rule="evenodd"
              d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
      <p
        class="text-sm text-slate-500"
      >
        Free shipping on all continental US orders.
      </p>
    </form>
  </div>
</div>
`;
